rust/lance-namespace/src/dir.rs

```diff
@@ -0,0 +1,674 @@
+        if let Some(id) = id {
+            if !id.is_empty() {
+                return Err(NamespaceError::InvalidConfiguration(
+                    format!("Directory namespace only supports root namespace operations, but got namespace ID: {:?}. Expected empty ID.", id)
+                ));
+            }
+        }
+        Ok(())
+    }
+    
+    fn get_table_path(&self, table_name: &str) -> String {
+        format!("{}.lance", table_name)
+    }
+    
+    fn convert_json_arrow_schema_to_arrow(&self, json_schema: &JsonArrowSchema) -> Result<Schema> {
+        let mut fields = Vec::new();
+        
+        for json_field in &json_schema.fields {
+            let arrow_type = self.convert_json_arrow_type_to_arrow(&json_field.r#type)?;
+            let nullable = json_field.nullable;
+            let field = Field::new(&json_field.name, arrow_type, nullable);
+            fields.push(field);
+        }
+        
+        let metadata = json_schema.metadata.as_ref()
+            .map(|m| m.iter().map(|(k, v)| (k.clone(), v.clone())).collect())
+            .unwrap_or_default();
+        
+        Ok(Schema::new_with_metadata(fields, metadata))
+    }
+    
+    fn convert_json_arrow_type_to_arrow(&self, json_type: &JsonArrowDataType) -> Result<DataType> {
+        match json_type.r#type.to_lowercase().as_str() {
+            "null" => Ok(DataType::Null),
+            "bool" | "boolean" => Ok(DataType::Boolean),
+            "int8" => Ok(DataType::Int8),
+            "uint8" => Ok(DataType::UInt8),
+            "int16" => Ok(DataType::Int16),
+            "uint16" => Ok(DataType::UInt16),
+            "int32" => Ok(DataType::Int32),
+            "uint32" => Ok(DataType::UInt32),
+            "int64" => Ok(DataType::Int64),
+            "uint64" => Ok(DataType::UInt64),
+            "float32" => Ok(DataType::Float32),
+            "float64" => Ok(DataType::Float64),
+            "utf8" => Ok(DataType::Utf8),
+            "binary" => Ok(DataType::Binary),
+            _ => Err(NamespaceError::InvalidConfiguration(format!("Unsupported Arrow type: {}", json_type.r#type))),
+        }
+    }
+}
+
+#[async_trait]
+impl LanceNamespace for DirNamespace {
+    async fn create_namespace(&self, _request: CreateNamespaceRequest) -> Result<CreateNamespaceResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support creating namespaces".to_string()
+        ))
+    }
+    
+    async fn list_namespaces(&self, _request: ListNamespacesRequest) -> Result<ListNamespacesResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support listing namespaces".to_string()
+        ))
+    }
+    
+    async fn describe_namespace(&self, _request: DescribeNamespaceRequest) -> Result<DescribeNamespaceResponse> {
+        Err(NamespaceError::NotSupported(
```

Comment:
this should return result if describing root namespace, and fail for other ones.